scala/scala-impl/src/org/jetbrains/sbt/project/structure/MillPreImporter.scala

```diff
@@ -0,0 +1,29 @@
+package org.jetbrains.sbt.project.structure
+
+import java.io.File
+import java.util.concurrent.atomic.AtomicBoolean
+import scala.concurrent.duration._
+import scala.collection.JavaConverters._
+
+object MillPreImporter {
+  private val MillProcessCheckTimeout = 100.millis
+
+  def setupBsp(baseDir: File): MillPreImporter = {
+    val processBuilder = new ProcessBuilder(Seq("./mill", "-i", "mill.contrib.BSP/install").asJava)
```

Comment:
Does that mean, one has to provide a mill wrapper script with name `mill` in the project? I'd rather not enforce this.

Reply:
With the current implementation, it's necessary for the automatic preimport. If the script is absent, it will not run and require manual invocation instead. We could avoid this by bundling a standalone Mill launcher, like we do with bloop or sbt, or if Mill can be fully bootstrapped from Coursier, which we also bundle. IMO doing it via Coursier would be my preferred option.

Reply:
That was a quick win implementation. I think the coursier route sounds way better.